perllib/FixMyStreet/Cobrand/Smidsy.pm

```diff
@@ -416,6 +416,307 @@ sub reports_hook_restrict_bodies_list {
     }
 }
 
+sub loop_period {
+    my ($date, $period, $end, $extra) = @_;
+    $end = FixMyStreet->set_time_zone($end)->truncate(to => $period)->add($period . 's' => 1)->subtract(seconds => 1);
+    my @out;
+    while ($date <= $end) {
+        push @out, { n => $date->$period, $extra ? (d => $date->$extra) : () };
+        $date->add($period . 's' => 1);
+    }
+    return @out;
+}
+
+sub get_body_children {
+    my ($self, $body) = @_;
+
+    $body = $self->{c}->model('DB::Body')->find( { id => $body } );
+    my $children = $body->first_area_children;
+    $self->{c}->stash->{area_body_name} = $body->name;
+    return [ values %$children ];
+}
+
+sub report_page_data {
+    my $self = shift;
+    my $c = $self->{c};
+
+    if (my $body = $c->get_param('body')) {
```

Comment:
This block is also in core, but after this fn is called. Could it be moved in core to be before this block is called, so it's not needed to be repeated here?